docs/src/dev-assess.md

```diff
@@ -0,0 +1,181 @@
+The only difference between 'scan' and 'regular' assess is how the packages built are located.
+All versions of assess produce the same output and metrics.
+Assess will normally build just like `cargo kani` or `cargo build`, whereas `scan` will find all cargo packages beneath the current directory, even in unrelated workspaces.
+Thus, 'scan' may be helpful in the case where the user has a choice of packages and is looking for the easiest to get started with (in addition to the Kani developer use-case, of aggregating statistics across many packages).
+
+(Tip: Assess may need to run for awhile, so try using `screen`, `tmux` or `nohup` to avoid terminating the process if, for example, an ssh connection breaks.)
+
+## What assess does
+
+Assess builds all the packages requested in "test mode" (i.e. `--tests`), and runs all the same tests that `cargo test` would, except through Kani.
+This gives end-to-end assurance we're able to actually build and run code from these packages, skipping nothing of what the verification process would need, except that the harnesses don't have any nondeterminism (`kani::any()`) and consequently don't "prove" much.
+The interesting signal comes from what tests cannot be analyzed by Kani due to unsupported features, performance problems, crash bugs, or other issues that get in the way.
+
+Currently, assess forces termination by using `unwind(1)` on all tests, so many tests will fail with unwinding assertions.
+
+## Current Assess Results
+
+Assess produces a few tables of output (both visually in the terminal, and in a more detailed json format) so far:
+
+### Unsupported features
+
+```text
+======================================================
+ Unsupported feature           |   Crates | Instances
+                               | impacted |    of use
+-------------------------------+----------+-----------
+ caller_location               |       71 |       239
+ simd_bitmask                  |       39 |       160
+...
+```
+
+The unsupported features table aggregates information about features that Kani does not yet support.
+These correspond to uses of `codegen_unimplemented` in the `kani-compiler`, and appear as warnings during compilation.
+
+Unimplemented features are not necessarily actually hit by (dynamically) reachable code, so an immediate future improvement on this table would be to count the features *actually hit* by failing test cases, instead of just those features reported as existing in code by the compiler.
+In other words, the current unsupported features table is **not** what we'd really want to see, in order to actually prioritize implementing these features, because we may be seeing a lot of features that won't actually "move the needle" in making it more practical to write proofs.
+Because of our operating hypothesis that code covered by tests is code that could be covered by proof, measuring unsupported features by those actually hit by a test should provide a better "signal" about priorities.
```

Comment:
> I don't know if I agree here. Conceptually, finding a failure that no existing unit test finds is exploring uncovered code.

Right, but we aren't finding failures that no existing unit test covers?

> Also, wouldn't this already be reported in the test failure table?

The test failure table is largely _expected_ to see tests fail because of deficiencies in Kani, not something else. (We could conceivably find more failures, like MIRI does, but we don't implement a lot of Rust-specific undefined behavior yet, so this is less likely.)

Reply:
I mean the runtime check. Wouldn't that be the reason why the test failed? Or is this reported as an assertion failure today?

Reply:
For an unsupported feature or missing function that we hit, yeah, it comes back as a failed property. (Not necessarily with the 'assertion' property class.)